File: week1/TugceErdemliAlSolWeek1/ValidAnagram.py
class Solution(object):
    def isAnagram(self, s, t):
        """
        :type s: str
        :type t: str
        :rtype: bool
        """

        if len(s) == len(t):
            for i in s:
                if i in t:
                    t = t.replace(i, '', 1)
                else:
                    return False
        else:
            return False
        return True

File: week1/esadmsk/valid-anagram.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if (s.Length != t.Length)
            return false;
        
        Dictionary<char, int> charCount = new Dictionary<char, int>();
        
        foreach (char c in s) {
            if (charCount.ContainsKey(c))
                charCount[c]++;
            else
                charCount[c] = 1;
        }
        
        foreach (char c in t) {
            if (!charCount.ContainsKey(c) || charCount[c] == 0)
                return false;
            
            charCount[c]--;
        }
        
        return true;
    }
}
 
